lib/miq_automation_engine/engine/miq_ae_event.rb

```diff
@@ -121,28 +109,52 @@ def self.process_result(ae_result, aevent)
   rescue URI::InvalidURIError => err
   end
 
-  def self.automate_user
-    # TODO: Get the real userid to pass into Automate
-    user = User.where(:userid => "admin").first
-    raise "Admin user needed to raise events" unless user
-    user
-  end
-
-  def self.call_automate(obj_type, obj_id, attrs, instance_name,
-                         user, state = nil, message = nil)
+  def self.call_automate(obj, attrs, instance_name, message = nil)
+    user_id, group_id, tenant_id = automate_user_ids(obj)
     args = {
-      :object_type      => obj_type,
-      :object_id        => obj_id,
+      :object_type      => obj.class.name,
+      :object_id        => obj.id,
       :attrs            => attrs,
       :instance_name    => instance_name,
-      :user_id          => user.id,
-      :miq_group_id     => user.current_group.id,
-      :tenant_id        => user.current_tenant.id,
-      :automate_message => message,
-      :state            => state
+      :user_id          => user_id,
+      :miq_group_id     => group_id,
+      :tenant_id        => tenant_id,
+      :automate_message => message
     }
+
     MiqAeEngine.deliver(args)
   end
 
-  private_class_method :automate_user, :call_automate
+  def self.provider_event_target(event)
+    event.vm_or_template || event.ext_management_system
+  end
+
+  def self.miq_event_target(target)
+    case target
+    when VmOrTemplate, MiqServer
+      target
+    else
+      target.ext_management_system
+    end
+  end
+
+  def self.automate_user_ids(object)
+    target = case object
+             when EmsEvent
+               provider_event_target(object)
+             else
+               miq_event_target(object)
+             end
+
+    user  = User.super_admin
+    group = target.kind_of?(MiqServer) ? user.current_group : target.miq_group
+    raise "A group is needed to raise events" unless group
+
+    tenant = group.current_tenant
+    user = target.evm_owner if target.kind_of?(VmOrTemplate) && target.evm_owner && target.evm_owner.current_group == target.miq_group
```

Comment:
@lfu The condition at the end should be using `group` instead of `target.miq_group`.  Also, I think the last condition should be checking if `target.evm_owner.miq_groups.include?(group)`, not just current_group.  Correct?